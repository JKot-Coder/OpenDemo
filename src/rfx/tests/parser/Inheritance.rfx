// :PARSER

Object: {
    Name: "Unknown",
    Valid: true,
},

Animal: {
    Weight: -1.0,
    Type: "None"
},

Animals: {
    Fox: {
        << [Object, Animal],
        Name: "Fox",
        Weight: 7.5,
        Type: "Predator",
    },

    Cow: {
        << [Object, Animal],
        Name: "Cow",
        Weight: 750,
        Type: "Pray",
    },

    BlackCow: { << Cow, Name: "BlackCow" },
},

Zoo: {
    << Object,
    Name: "Park",
    Animals:[
        { << Animals.BlackCow }, { << Animals.Fox, Name: "RedFox" }
    ],
}
