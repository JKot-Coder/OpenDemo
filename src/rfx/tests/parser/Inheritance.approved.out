Diagnostic:[
]
SourceOutput:[
    {
      "Animal": {
        "Type": "None",
        "Weight": -1
      },
      "Animals": {
        "BlackCow": {
          "Name": "BlackCow",
          "Type": "Pray",
          "Valid": true,
          "Weight": 750
        },
        "Cow": {
          "Name": "Cow",
          "Type": "Pray",
          "Valid": true,
          "Weight": 750
        },
        "Fox": {
          "Name": "Fox",
          "Type": "Predator",
          "Valid": true,
          "Weight": 7.5
        }
      },
      "Object": {
        "Name": "Unknown",
        "Valid": true
      },
      "Zoo": {
        "Animals": [
          {
            "Name": "BlackCow",
            "Type": "Pray",
            "Valid": true,
            "Weight": 750
          },
          {
            "Name": "RedFox",
            "Type": "Predator",
            "Valid": true,
            "Weight": 7.5
          }
        ],
        "Name": "Park",
        "Valid": true
      }
    }
]
